src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -3568,16 +3588,30 @@ def _sync_runs(self) -> None:
         from zenml.zen_stores.migrations.alembic import AlembicVersion
 
         try:
-            with Session(self.engine) as session:
-                # We use the alembic version table as a shared resource that we
-                # can lock to prevent multiple processes from syncing runs at
-                # the same time.
-                logger.debug("Syncing pipeline runs...")
-                session.query(AlembicVersion).with_for_update().all()
-                self._sync_runs_with_lock(session)
-                logger.debug("Pipeline runs sync complete")
+            # This is to synchronize the locally running threads so that only
+            # one thread attempts to sync the runs at any given time. The
+            # timeout is to ensure that we don't block the threads for too long.
+            if self.sync_lock.acquire(timeout=30):
```

Comment:
Can we make the timeout shorter? Perhaps this could also improve our performance issues since if there are a lot of requests trying to sync at the same time this should then only allow one of them to do the actual sync and skip syncing for the others, right?

Reply:
yes, good idea, I will make it 20 seconds. That way, the client won't have time to timeout (default client timeout is 30 seconds now)